openml-lightgbm/openml-lightgbm-provider/src/main/java/com/feedzai/openml/provider/lightgbm/SWIGTrainResources.java

```diff
@@ -93,7 +93,7 @@ public SWIGTrainResources(final int numInstances, final int numFeatures) {
 
         logger.debug("Allocating SWIG train data array.");
         // 1-D Array in row-major-order that stores only the features (excludes label) in double format:
-        this.swigTrainFeaturesDataArray = lightgbmlib.new_doubleArray(numInstances * numFeatures);
+        this.swigTrainFeaturesDataArray = lightgbmlib.new_doubleArray(((long)numInstances) * ((long)numFeatures));
```

Comment:
Isn't it better to change the parameters of this method to long?

Reply:
I thought the same first, but LightGBM's Java API requires an int for those 2 parameters everywhere.
So it's best not to use a larger data type.

Reply:
Basically only our train buffer size can be a long. For LightGBM's datasets, both dimensions are limited to ints.

I don't like the "look" of the code though, is there a more "idiomatic" way to write it? I'm afraid of removing the double parenthesis as it might not cast before the product.